src/ds/champ_map.h

```diff
@@ -126,8 +126,10 @@ namespace champ
         const auto& entry = bin[i];
         if (k == entry->key)
         {
+          const auto diff = map::get_serialized_size_with_padding(entry->key) +
+            map::get_serialized_size_with_padding(entry->value);
           bin[i] = std::make_shared<Entry<K, V>>(k, v);
-          return map::get_size(k) + map::get_size(v);
+          return diff;
```

Comment:
This is the actual fix. In the unlikely case that we have a collision (this is in `struct Collisions`), we used to return the serialised size of the new (k,v), rather than the old size if that key existed.